packages/sessions-sdk-react/src/components/renew-session-modal.tsx

```diff
@@ -0,0 +1,113 @@
+import { useCallback } from "react";
+import { Heading } from "react-aria-components";
+
+import type { EstablishedSessionState } from "../session-state.js";
+import { ModalDialog } from "./modal-dialog.js";
+import styles from "./renew-session-modal.module.css";
+import { SessionLimits } from "./session-limits.js";
+import { Spinner } from "./spinner.js";
+import { useSessionContext, useSession } from "../hooks/use-session.js";
+import {
+  useTokenAccountData,
+  StateType as TokenDataStateType,
+} from "../hooks/use-token-account-data.js";
+import { StateType } from "../session-state.js";
+
+export const RenewSessionModal = () => {
+  const sessionState = useSession();
+
+  const onOpenChange = useCallback(
+    (isOpen: boolean) => {
+      if (!isOpen && "cancel" in sessionState) {
```

Comment:
is there a way to expose `isCanceling` or some other method on `sessionState` to avoid querying for strings?

Reply:
There is but it's entirely redundant, we're not actually querying for strings here, this code is saying "Is the session state in any state that has a `cancel` function?" -- i.e. it would be equivalent but redundant (and thus error prone) to add a flag `canCancel` or have a filter that explicitly enumerates the states that can be canceled.  IMO the implementation here is the simplest -- we cancel the action if the action can be canceled, which is indicated by the existence of a `cancel` method in the state.  If we add or remove a cancel method down the line to a state, the UI will automatically work or prevent the modal from closing.